engine/src/main/java/org/operaton/bpm/engine/impl/util/JsonUtil.java

```diff
@@ -684,32 +684,32 @@ public static Gson getGsonMapper() {
 
   public static Gson createGsonMapper() {
     return new GsonBuilder()
-      .serializeNulls()
-      .registerTypeAdapter(Map.class, (JsonDeserializer<Map<String, Object>>) (json, typeOfT, context) -> {
-
-        Map<String, Object> map = new HashMap<>();
-
-        for (Map.Entry<String, JsonElement> entry : getObject(json).entrySet()) {
-          if (entry != null) {
-            String key = entry.getKey();
-            JsonElement jsonElement = entry.getValue();
-
-            if (jsonElement != null && jsonElement.isJsonNull()) {
-              map.put(key, null);
-
-            } else if (jsonElement != null && jsonElement.isJsonPrimitive()) {
-
-              Object rawValue = asPrimitiveObject((JsonPrimitive) jsonElement);
-              if (rawValue != null) {
-                map.put(key, rawValue);
-
-              }
-            }
-          }
-        }
-
-        return map;
-      })
-      .create();
+        .serializeNulls()
+        .registerTypeAdapter(Map.class, (JsonDeserializer<Map<String, Object>>) (json, typeOfT, context) -> {
+          Map<String, Object> map = new HashMap<>();
+
+          getObject(json)
+              .entrySet()
+              .stream()
+              .filter(entry -> entry != null && entry.getValue() != null)
```

Comment:
entry can't be null, right? Did you want to check the key for not null?
effective, value can't be null, too, but this requires knowledge about the JSON parsing